ui/shortcuts/src/main/java/org/cru/godtools/shortcuts/GodToolsShortcutManager.kt

```diff
@@ -267,6 +257,11 @@ class GodToolsShortcutManager @VisibleForTesting internal constructor(
     }
     // endregion Update Existing Shortcuts
 
+    internal suspend fun refreshShortcutsNow() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) updateShortcuts()
+        updatePendingShortcuts()
+    }
+
```

Comment:
What is the checking against exactly in order to know if it should refresh the shortcuts immediately?

Reply:
this method allows us to trigger an immediate refresh of all shortcuts. right now the only place this is triggered from is the `UpdateShortcutsWorker`, which is only scheduled after the user changes their device language.

The reason we have this execution path enabled is because we use the device language to translate the tool names for shortcuts, so when the device language changes we need to update the shortcuts to use the new device language.

Reply:
Originally the system locale update broadcast would directly do this work and not rely on work manager to schedule it, but there are timing constraints that come with updating immediately while processing the broadcast.

Broadcast Receivers start execution on the main thread, they can offload processing to a background thread with `goAsync()`, but they additionally have a 10 second time limit to complete execution during. The recommended solution for any heavy background processing is to use Work Manager to schedule the work to be done at a later time, with assurances that it will fully execute at some point in the future.